Bright Tides/Assets/Scripts/Controllers/EnemyController.cs

```diff
@@ -0,0 +1,171 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+public class EnemyController: MonoBehaviour {
+
+    public List<Entity> enemies; // The list of all enemies in the region
+    private Tile[,] tileMap;
+
+    private bool isPerformingActions = false; // The flag to track if the controller is currently scheduling or performing actions
```

Comment:
I moreso saw this as the 'enemy is performing action' in my state machine drawing. I think it's just badly named because it is supposed to be distinct from the one in the game manager.

Although, maybe it could share that one, I just wanted to be safe